### PR TITLE
CLOUDP-276968: improve whoami err message

### PR DESCRIPTION
**Pull request by tibulca**
## Proposed changes
improve whoami err message

_Jira ticket:_ CLOUDP-276968

### CODE REVIEW

**Review by andreaangiolillo — DISMISSED** [review]

File: internal/cli/auth/whoami.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func (opts *whoOpts) Run() error {
 	return nil
 }
 
-var ErrUnauthenticated = errors.New("not logged in")
+var ErrUnauthenticated = errors.New("not logged in with an Atlas account")
 
 func AccountWithAccessToken() (string, error) {
```
Comment (gssbzn) [inline internal/cli/auth/whoami.go:40]: what about
```suggestion
func AccountWithAccessToken() (string, error) {
	if config.PublicAPIKey() != "" {
		return config.PublicAPIKey(), nil
	}
	if config.AccessToken() != "" {
		return config.AccessTokenSubject()
	}
	return  "", ErrUnauthenticated
}
```

**Review by gssbzn — APPROVED** [review]
LGTM feel free to double check the copy with docs

File: internal/cli/auth/whoami.go (hunk 1 above)
Comment (fmenezes) [reply internal/cli/auth/whoami.go:40]: if we are doing this shouldn't we say `Logged in as email@server.com` and `Logged in as public key XXX` as opposed to `Logged in as XXX`

Comment (fmenezes) [reply internal/cli/auth/whoami.go:40]:
```suggestion
func AccountWithAccessToken() (string, string, error) {
	if config.PublicAPIKey() != "" {
		return config.PublicAPIKey(), "key", nil
	}
	if config.AccessToken() != "" {
		return config.AccessTokenSubject(), "account"
	}
	return  "", "", ErrUnauthenticated
}
```

Comment (tibulca) [reply internal/cli/auth/whoami.go:41]: makes sense to add a bit more context as I also realized that AccountWithAccessToken() is re-used in the setup cmd so I split the logic too

**Review by fmenezes — APPROVED** [review]

**Review by corryroot — APPROVED** [review]
LGTM!

**Pull request merged**